packages/app/dashboard/service.ts

```diff
@@ -0,0 +1,125 @@
+import type { User, UserRole } from 'auth/types'
+import { getLoggedInUser } from 'auth/service'
+import type { IncomingMessage, ServerResponse } from 'http'
+import type { Model, ModelCategory } from 'models/types'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { getDocumentsForModel } from 'documents/service'
+import { getCommentsForUser } from 'comments/service'
+
+const collator = new Intl.Collator()
+
+async function getRecentDocumentsFromModels(modelNames: string[]) {
+    const documentPromises = []
+    const recentDocuments = []
+
+    for (const name of modelNames) {
+        documentPromises.push(getDocumentsForModel(name))
+    }
+
+    const results = await Promise.allSettled(documentPromises)
+
+    for (const result of results) {
+        if (result.status === 'fulfilled') {
+            const [_error, documents] = result.value
+
+            //! More useful would be getting "n" documents by status. Knowing what statuses (which are dynamic) to use is currently a blocker.
```

Comment:
getRecentDocumentForModels is a little weird, because what we really want is something like a paginated list of documents (or n documents—I'm not sure if dashboard modules should be full-featured enough to have pagination or not), sorted by date, arranged by status, which is what we show in the UI. Slicing at 30 is a first attempt, but we should probably refine this later. We're balancing page load time with content.